flink-tests/src/test/java/org/apache/flink/api/connector/source/lib/IteratorSourcesITCase.java

```diff
@@ -71,6 +85,36 @@ public void testParallelSourceExecution() throws Exception {
         verifySequence(result, 1L, 1_000L);
     }
 
+    @Test
+    public void testCheckpointingWithDelayedAssignment() throws Exception {
+        StreamExecutionEnvironment env = StreamExecutionEnvironment.getExecutionEnvironment();
+        env.setParallelism(1);
+        env.setRestartStrategy(RestartStrategies.noRestart());
+        env.enableCheckpointing(10, CheckpointingMode.EXACTLY_ONCE);
+        final SingleOutputStreamOperator<Long> stream =
+                env.fromSequence(0, 100)
+                        .map(
+                                x -> {
+                                    Thread.sleep(10);
```

Comment:
10ms is pretty long here, would 2ms also do it?

Reply:
I have slighty increased it instead, but only apply it to the first element now.